bson/src/test/unit/org/bson/types/ObjectIdTest.java

```diff
@@ -58,6 +59,33 @@ public void testBytes() {
     }
 
     @Test
+    public void testByteBuffer() {
```

Comment:
I think it would be clearer if, instead of adding a new test, you just add another assertion to testToByteArray and to testFromByteArray (perhaps renaming to testToBytes and testFromBytes)